### PR TITLE
fix: [CI-6798]: open run pipeline modal for re-run in debug mode

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

open run pipeline modal for re-run in debug mode

#### Screenshots

Screenshot 2023-01-23 at 6 34 09 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by shaurya-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx [hunk 1]
```diff
@@ -87,6 +88,21 @@ export function ExecutionHeader({ onRunPipelineInDebugMode }: ExecutionHeaderPro
...
+    branch,
+    connectorRef,
+    storeType,
+    stagesExecuted: pipelineExecutionSummary?.stagesExecuted,
+    isDebugMode: hasCI
+  })
+
+  const handleRunPipelineInDebugMode = () => {
+    openRunPipelineModal()
+  }
```
Comment (d1wilko) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:105]: is this needed?

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx [hunk 2]
```diff
@@ -191,7 +207,7 @@ export function ExecutionHeader({ onRunPipelineInDebugMode }: ExecutionHeaderPro
             canExecute={canExecute}
             canRetry={pipelineExecutionSummary.canRetry}
             modules={pipelineExecutionSummary.modules}
-            onReRunInDebugMode={hasCI ? () => onRunPipelineInDebugMode() : undefined}
+            onReRunInDebugMode={hasCI ? () => handleRunPipelineInDebugMode() : undefined}
```
Comment (d1wilko) [inline src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx:210]: can we just call `openRunPipelineModal ()` here?

**Review by d1wilko — DISMISSED** [review]
no major concerns - one suggested change but I'll leave it up to you :)

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx]: no, removing this

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx (hunk 2 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionHeader/ExecutionHeader.tsx]: right, updated

**Comment by shaurya-harness** [comment]
retrigger jest

**Comment by shaurya-harness** [comment]
retrigger coverage

**Review by d1wilko — APPROVED** [review]
LGTM :)

**Comment by shaurya-harness** [comment]
retrigger jest

**Review by vkbansal-harness — APPROVED** [review]

**Comment by shaurya-harness** [comment]
retrigger jest

**Pull request merged**